### PR TITLE
operator: add github workflow in ci

### LINKED ISSUE #5: [Build] Add GitHub Workflow for CI
This repo will contain a Kubernetes Operator based on the Operator SDK using Golang.

Docker Repo: https://hub.docker.com/repository/docker/layer5/meshery-operator

### PR DESCRIPTION
**Pull request by subhamkrai**
this commit adds github actions in ci.

Signed-off-by: subhamkrai <subhamkumarrai03@gmail.com>

**Description**

This PR fixes #5 

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by subhamkrai** [comment]
@leecalcote let me know if this is what we want. I just copied from [here](https://github.com/layer5io/meshery-adapter-template/blob/master/.github/workflow/ci.yml)

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,148 @@
+name: Meshery Operator Template
```
Comment (PluckyPrecious) [inline .github/workflows/ci.yaml:1]: I think the name should be Meshery Operator

**Review by PluckyPrecious — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (subhamkrai) [reply .github/workflows/ci.yaml]: Done. Thanks!

**Comment by subhamkrai** [comment]
PTAL

**Comment by leecalcote** [comment]
@kumarabd Will this workflow work?

**Comment by leecalcote** [comment]
> @kumarabd Will this workflow work?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+      with:
+        fetch-depth: 1
+    - name: Setup Go
+      uses: actions/setup-go@v1
+      with:
+        go-version: ${{ secrets.GO_VERSION }}
+    - run: GOPROXY=direct GOSUMDB=off GO111MODULE=on go build -tags draft ./cmd/main.go
+    - uses: actions/setup-node@v1
+      with:
+        node-version: ${{ secrets.NODE_VERSION }}
```
[the remark this answers is not in the document]
Comment (subhamkrai) [reply .github/workflows/ci.yaml:108]: @kumarabd  by `these` you mean the `L107-L108` or the `build` workflow? Thanks for review

[the remark this answers is not in the document]
Comment (kumarabd) [reply .github/workflows/ci.yaml:108]: Yeah basically we don't have a nodejs dependency in the project

[the remark this answers is not in the document]
Comment (subhamkrai) [reply .github/workflows/ci.yaml]: Done.

**Comment by leecalcote** [comment]
@kushthedude will you review?

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    steps:
+    - name: Check out code
+      uses: actions/checkout@master
+      with:
+        fetch-depth: 1
+    - name: Setup Go
+      uses: actions/setup-go@v1
+      with:
+        go-version: ${{ secrets.GO_VERSION }}
+    - run: GOPROXY=direct GOSUMDB=off GO111MODULE=on go build -tags draft ./cmd/main.go
```
Comment (kushthedude) [inline .github/workflows/ci.yaml:105]: Can location be corrected for the main.go?

**Review by kushthedude — COMMENTED** [review]
looks good for first pass

File: .github/workflows/ci.yaml (hunk 3 above)
Comment (subhamkrai) [reply .github/workflows/ci.yaml]: Done.

**Comment by leecalcote** [comment]
@kushthedude @subhamkrai @kalradev does this work?

**Comment by subhamkrai** [comment]
> @kushthedude @subhamkrai @kalradev does this work?

@leecalcote I see `error check` and `static check` is working(showing errors) but for `security check`, `review code` and `vet` is failing due to modules we have.

**Comment by leecalcote** [comment]
@kushthedude, @kalradev will reach out to you on this...

**Comment by kushthedude** [comment]
Superseded by #79

**Pull request closed without merging**